src/core/lib/iomgr/tcp_server_uv.c

```diff
@@ -142,10 +147,12 @@ static void handle_close_callback(uv_handle_t *handle) {
 }
 
 static void close_listener(grpc_tcp_listener *sp) {
+  grpc_exec_ctx exec_ctx = GRPC_EXEC_CTX_INIT;
   if (!sp->closed) {
     sp->closed = true;
     uv_close((uv_handle_t *)sp->handle, handle_close_callback);
   }
+  grpc_exec_ctx_finish(&exec_ctx);
```

Comment:
Does `uv_close` call its callback immediately potentially so you had to add a call to `grpc_exec_ctx_finish` here ?